crates/chain/src/indexed_tx_graph.rs

```diff
@@ -233,7 +233,18 @@ pub trait Indexer {
     /// Scan and index the given `outpoint` and `txout`.
     fn index_txout(&mut self, outpoint: OutPoint, txout: &TxOut) -> Self::ChangeSet;
 
-    /// Scan and index the given transaction.
+    /// Scans a transaction for relevant outpoints, which are stored and indexed internally.
+    ///
+    /// If the matched script pubkey is part of the lookahead, the last stored index is updated for
+    /// the script pubkey's keychain and the [`ChangeSet`] returned will reflect the
+    /// change.
```

Comment:
lookahead has nothing to do with this trait. That's an API of `KeychainTxOutIndex`. I think this documentation is missing the point (probably because it was copy pasted) for this methid. The documentation of the trait should mention that it's primarily used by `IndexedTxGraph`. Then we can explain here that `index_tx` is called when there's a new transaction added to the graph.